fs/operations/operations.go

```diff
@@ -91,6 +92,26 @@ func checkHashes(ctx context.Context, src fs.ObjectInfo, dst fs.Object, ht hash.
 	if dstHash == "" {
 		return true, hash.None, srcHash, dstHash, nil
 	}
+	srcHashes := strings.Split(srcHash, "-")
+	dstHashes := strings.Split(dstHash, "-")
+	if len(srcHashes) == 2 && len(dstHashes) == 2 && srcHashes[1] != dstHashes[1] {
+		fs.Debugf(src, "chunk size differ")
+		return true, hash.None, "", "", nil
```

Comment:
This chunk is located in a generic function working with backends of all types and hashsums of all types.
Hashsum here is an **opaque blob of bytes** that can be equal or not.
You can't treat it as having any internal structure at this level.

Reply:
This chunk should be factored into a separate function and activated by a non-backend runtime flag.

Reply:
This chunk should be factored into a separate function. You will call it from this place if enabled by a new non-backend flag.

Reply:
ok

Reply:
Another solution that frees user from the need to add **two** flags:
Create a generic `Fs` interface, e.g. call it `MultipartHashComparer` with a single function `CompareMultipartHashes` and add it solely to `Object` returned by `S3`. At this point in _operations.go_ you will do something like
```
if mhc, ok := src.(MultipartHashComparer); ok {
  result, err := src.(MultipartHashComparer).CompareMultipartHashes(dst)
 .........
}
```

Reply:
@kmlebedev @ncw 
Do you think it's worth an effort?

Reply:
Another solution that frees user from the need to add **two** flags:
Create a generic `Fs` interface, e.g. call it `MultipartHashComparer` with a single function `CompareMultipartHashes` and add it solely to `Object` returned by `S3`. At this point in _operations.go_ you will do something like
```
if mhc, ok := src.(MultipartHashComparer); ok {
  result, err := src.(MultipartHashComparer).CompareMultipartHashes(dst)
 .........
}
```

@ncw @kmlebedev
Do you think it's worth the effort?

Reply:
Actually I see why it can be **worth** the effort.
You will have to add a test for your new feature anyway!
The test procedure for special-case provider requirements is usually `Take such'n'such remote with such'n'such config and run integration test again it`. Well... An S3 flag fits fine in this scenario, but a non-backend flag doesn't. For generic flags we usually write a new test case in `fs/operatations_test.go`. If such a test requires a specific type of provider, we add a skip condition. What can be required in context of this change??
I hope I'm clear enough.